src/components/Search/Search.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { MoviesList } from '../MoviesList';
+
+export class Search extends React.Component {
+  state = {
+    query: '',
+  }
+
+  render() {
+    const { moviesList } = this.props;
+    const { query } = this.state;
+    let filteredMoveisList = [];
+
+    if (query) {
+      const searchPhrase = query.toLocaleLowerCase();
+
+      filteredMoveisList = moviesList.filter(({ title, description }) => (
+        title.toLocaleLowerCase().includes(searchPhrase)
+            || description.toLocaleLowerCase().includes(searchPhrase)
+      ));
+    }
```

Comment:
```suggestion
    const filteredMoveisList = moviesList.filter(({ title, description }) => (
        title.toLocaleLowerCase().includes(query)
            || description.toLocaleLowerCase().includes(query)
      ));
```
У тебя много логики базируется на том, чтобы в то время как мы ничего не ввели (query = ' '), фильмы отобразились полностью. Но если в метод includes() положить пустую строку и искать в стринговой поле, он всегда вернет true. Поэтому при пустой строке, у нас выведутся все фильмы. Нет потребности в этом if. И внизу при передаче, можно всегда передавать отфильтрованный массив.

Reply:
WOW
cool comment
thanks